packages/netlify-cms-widget-markdown/src/MarkdownControl/VisualEditor.js

```diff
@@ -45,249 +83,116 @@ export default class Editor extends React.Component {
+    onChange(markdown);
   }, 150);
 
-  handleChange = change => {
-    if (!this.state.value.document.equals(change.value.document)) {
-      this.handleDocumentChange(change);
+  handleChange = editor => {
+    if (!this.state.value.document.equals(editor.value.document)) {
+      this.handleDocumentChange(editor);
     }
-    this.setState({ value: change.value });
+    this.setState({ value: editor.value });
   };
 
   processRef = ref => {
-    this.ref = ref;
+    this.editor = ref;
   };
 
   render() {
-    const { onAddAsset, getAsset, className, field, getEditorComponents } = this.props;
-
+    const { onAddAsset, getAsset, className, field } = this.props;
     return (
-      <VisualEditorContainer>
+      <div
+        css={coreCss`
+          position: relative;
+        `}
+      >
         <EditorControlBar>
           <Toolbar
             onMarkClick={this.handleMarkClick}
             onBlockClick={this.handleBlockClick}
-            onLinkClick={this.handleLink}
-            selectionHasMark={this.selectionHasMark}
-            selectionHasBlock={this.selectionHasBlock}
-            selectionHasLink={this.hasLinks}
-            onToggleMode={this.handleToggle}
-            plugins={getEditorComponents()}
-            onSubmit={this.handlePluginAdd}
+            onLinkClick={this.handleLinkClick}
+            onToggleMode={this.handleToggleMode}
+            plugins={this.editorComponents}
+            onSubmit={this.handleInsertShortcode}
             onAddAsset={onAddAsset}
             getAsset={getAsset}
             buttons={field.get('buttons')}
+            hasMark={this.hasMark}
+            hasInline={this.hasInline}
+            hasBlock={this.hasBlock}
           />
         </EditorControlBar>
         <ClassNames>
           {({ css, cx }) => (
-            <Slate
+            <div
               className={cx(
                 className,
                 css`
                   ${visualEditorStyles}
                 `,
               )}
-              value={this.state.value}
-              renderNode={renderNode}
-              renderMark={renderMark}
-              validateNode={validateNode}
-              plugins={plugins}
-              onChange={this.handleChange}
-              onKeyDown={onKeyDown}
-              onPaste={this.handlePaste}
-              ref={this.processRef}
-              spellCheck
-            />
+            >
+              <Slate
+                className={css`
+                  padding: 16px 20px 0;
+                `}
+                value={this.state.value}
+                renderBlock={this.renderBlock}
+                renderInline={this.renderInline}
+                renderMark={this.renderMark}
+                schema={this.schema}
+                plugins={this.plugins}
+                onChange={this.handleChange}
+                ref={this.processRef}
+                spellCheck
+              />
+              <InsertionPoint onClick={this.handleClickBelowDocument} />
```

Comment:
Slate doesn't expand to fill available space, so we need to intelligently capture focus from clicks within the widget but below the editor.